src/configurator/question.rb

```diff
@@ -0,0 +1,128 @@
+
+# frozen_string_literal: true
+
+require 'active_support/core_ext/module/delegation'
+require 'highline'
+require 'patches/highline'
+
+HighLine::Question.prepend Metalware::Patches::HighLine::Question
+HighLine::Menu.prepend Metalware::Patches::HighLine::Menu
+
+module Metalware
+  class Configurator
+    class Question
+      def initialize(question_node)
+        @question_node = question_node
+        @highline = HighLine.new
+      end
+
+      attr_accessor :default, :old_answer, :progress_indicator
+
+      def ask
+        ask_method = choices.nil? ? "ask_#{type}_question" : 'ask_choice_question'
+        answer = send(ask_method) { |q| configure_question(q) }
+        answer.tap { |a| question_node.answer = a }
```

Comment:
AFAICS we don't need the `answer` variable and the `tap`ing - it would be clearer to combine these two lines like this:

```ruby
question_node.answer = send(ask_method) { |q| configure_question(q) }
```

Reply:
Condensed